Compiling without GDScript fails again

**Godot version:**
3.2 d773de6d244116c3ad26f0506a10038cc11019ff
(I didn't check this on the master branch)


**OS/device including version:**
Windows


**Issue description:**
Build without GDScript failed with tools=yes


**Steps to reproduce:**
```scons p=windows tools=yes gdscript=no module_gdscript_enabled=no```

[log_tools.txt](https://github.com/godotengine/godot/files/5163219/log_tools.txt)

Basically this according to the log:

```
[ 96%] Linking Static Library ==> modules\freetype\freetype_builtin.windows.tools.64.lib
[ 96%] Linking Program        ==> bin\godot.windows.tools.64.exe
   Creating library bin\godot.windows.tools.64.lib and object bin\godot.windows.tools.64.exp
editor.windows.tools.64.lib(plugin_config_dialog.windows.tools.64.obj) : error LNK2019: unresolved external symbol "public: __cdecl GDScript::GDScript(void)" (??0GDScript@@QEAA@XZ) referenced in function "private: void __cdecl PluginConfigDialog::_on_confirmed(void)" (?_on_confirmed@PluginConfigDialog@@AEAAXXZ)
editor.windows.tools.64.lib(plugin_config_dialog.windows.tools.64.obj) : error LNK2001: unresolved external symbol "private: static class GDScriptLanguage * GDScriptLanguage::singleton" (?singleton@GDScriptLanguage@@0PEAV1@EA)
bin\godot.windows.tools.64.exe : fatal error LNK1120: 2 unresolved externals
scons: *** [bin\godot.windows.tools.64.exe] Error 1120
scons: building terminated because of errors.
```

By the way, further regressions like this could be handled by #41091 and including that in CI.

--- a/editor/plugin_config_dialog.cpp
+++ b/editor/plugin_config_dialog.cpp
@@ -35,8 +35,11 @@
 #include "editor/editor_plugin.h"
 #include "editor/editor_scale.h"
 #include "editor/project_settings_editor.h"
-#include "modules/gdscript/gdscript.h"
 #include "scene/gui/grid_container.h"
+
+#ifdef GDSCRIPT_ENABLED
+#include "modules/gdscript/gdscript.h"
+#endif
 
 void PluginConfigDialog::_clear_fields() {
 	name_edit->set_text("");
@@ -75,6 +78,7 @@ void PluginConfigDialog::_on_confirmed() {
 		// TODO Use script templates. Right now, this code won't add the 'tool' annotation to other languages.
 		// TODO Better support script languages with named classes (has_named_classes).
 
+#ifdef GDSCRIPT_ENABLED
 		if (lang_name == GDScriptLanguage::get_singleton()->get_name()) {
 			// Hard-coded GDScript template to keep usability until we use script templates.
 			Ref<Script> gdscript = memnew(GDScript);
@@ -95,12 +99,15 @@ void PluginConfigDialog::_on_confirmed() {
 			ResourceSaver::save(script_path, gdscript);
 			script = gdscript;
 		} else {
+#endif
 			String script_path = path.plus_file(script_edit->get_text());
 			String class_name = script_path.get_file().get_basename();
 			script = ScriptServer::get_language(lang_idx)->get_template(class_name, "EditorPlugin");
 			script->set_path(script_path);
 			ResourceSaver::save(script_path, script);
+#ifdef GDSCRIPT_ENABLED
 		}
+#endif
 
 		emit_signal("plugin_ready", script.operator->(), active_edit->is_pressed() ? subfolder_edit->get_text() : "");
 	} else {
@@ -229,9 +236,11 @@ PluginConfigDialog::PluginConfigDialog() {
 	for (int i = 0; i < ScriptServer::get_language_count(); i++) {
 		ScriptLanguage *lang = ScriptServer::get_language(i);
 		script_option_edit->add_item(lang->get_name());
+#ifdef GDSCRIPT_ENABLED
 		if (lang == GDScriptLanguage::get_singleton()) {
 			default_lang = i;
 		}
+#endif
 	}
 	script_option_edit->select(default_lang);
 	grid->add_child(script_option_edit);

